[ICE]: `CTFE tried to evaluate type-const`

<!--
[31mICE[0m: Rustc ./a.rs '' 'thread 'rustc' ($TID) panicked at compiler/rustc_const_eval/src/const_eval/eval_queries.rs:408:5: 'CTFE tried to evaluate type-const: DefId(0:3 ~ a[1854]::CONST)'', 'thread 'rustc' ($TID) panicked at compiler/rustc_const_eval/src/const_eval/eval_queries.rs:408:5: 'CTFE tried to evaluate type-const: DefId(0:3 ~ a[1854]::CONST)''
File: /tmp/im/2/a.rs
-->

auto-reduced (treereduce-rust):
````rust
type const CONST: usize = 1;

fn main() {
    match CONST {
        CONST => 0,
        _ => 1,
    };
}

````

original:
````rust
#![feature(min_generic_const_args)]
type const CONST: usize = 1;

fn main() {
    match CONST {
        CONST => 0,
        _ => 1,
    };
}

````

Version information
````
rustc 1.97.0-nightly (99eed207b 2026-05-10)
binary: rustc
commit-hash: 99eed207b47aca1fec5c665531db8e948a92d0ca
commit-date: 2026-05-10
host: x86_64-unknown-linux-gnu
release: 1.97.0-nightly
LLVM version: 22.1.4

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_const_eval/src/const_eval/eval_queries.rs#L402-L414

Command:
`/home/matthias/.rustup/toolchains/alt-master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0658]: `type const` syntax is experimental
 --> /tmp/icemaker_global_tempdir.PBQWEtlTwUlv/rustc_testrunner_tmpdir_reporting.b60aCTUxNMeC/mvce.rs:1:1
  |
1 | type const CONST: usize = 1;
  | ^^^^^^^^^^
  |
  = note: see issue #132980 <https://github.com/rust-lang/rust/issues/132980> for more information
  = help: add `#![feature(min_generic_const_args)]` to the crate attributes to enable
  = note: this compiler was built on 2026-05-10; consider upgrading it if it is out of date

error[E0658]: top-level `type const` are unstable
 --> /tmp/icemaker_global_tempdir.PBQWEtlTwUlv/rustc_testrunner_tmpdir_reporting.b60aCTUxNMeC/mvce.rs:1:1
  |
1 | type const CONST: usize = 1;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #132980 <https://github.com/rust-lang/rust/issues/132980> for more information
  = help: add `#![feature(min_generic_const_args)]` to the crate attributes to enable
  = note: this compiler was built on 2026-05-10; consider upgrading it if it is out of date


thread 'rustc' (3053138) panicked at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_const_eval/src/const_eval/eval_queries.rs:408:5:
CTFE tried to evaluate type-const: DefId(0:3 ~ mvce[dcec]::CONST)
stack backtrace:
   0:     0x7f684627c15c - std[170a660505e6ad19]::backtrace_rs::backtrace::libunwind::trace
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/../../backtrace/src/backtrace/libunwind.rs:117:9
   1:     0x7f684627c15c - std[170a660505e6ad19]::backtrace_rs::backtrace::trace_unsynchronized::<std[170a660505e6ad19]::sys::backtrace::_print_fmt::{closure#1}>
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/../../backtrace/src/backtrace/mod.rs:66:14
   2:     0x7f684627c15c - std[170a660505e6ad19]::sys::backtrace::_print_fmt
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/sys/backtrace.rs:74:9
   3:     0x7f684627c15c - <<std[170a660505e6ad19]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[d8f1845231c5d568]::fmt::Display>::fmt
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/sys/backtrace.rs:44:26
   4:     0x7f684151fd8e - <core[d8f1845231c5d568]::fmt::rt::Argument>::fmt
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/core/src/fmt/rt.rs:152:76
   5:     0x7f684151fd8e - core[d8f1845231c5d568]::fmt::write
   6:     0x7f68462940cc - std[170a660505e6ad19]::io::default_write_fmt::<std[170a660505e6ad19]::sys::stdio::unix::Stderr>
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/io/mod.rs:621:11
   7:     0x7f68462940cc - <std[170a660505e6ad19]::sys::stdio::unix::Stderr as std[170a660505e6ad19]::io::Write>::write_fmt
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/io/mod.rs:1976:13
   8:     0x7f6846240f34 - <std[170a660505e6ad19]::sys::backtrace::BacktraceLock>::print
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/sys/backtrace.rs:47:9
   9:     0x7f6846240f34 - std[170a660505e6ad19]::panicking::default_hook::{closure#0}
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:292:27
  10:     0x7f684626b081 - std[170a660505e6ad19]::panicking::default_hook
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:319:9
  11:     0x7f68429759e8 - <alloc[eced2805efacec9a]::boxed::Box<dyn for<'a, 'b> core[d8f1845231c5d568]::ops::function::Fn<(&'a std[170a660505e6ad19]::panic::PanicHookInfo<'b>,), Output = ()> + core[d8f1845231c5d568]::marker::Send + core[d8f1845231c5d568]::marker::Sync> as core[d8f1845231c5d568]::ops::function::Fn<(&std[170a660505e6ad19]::panic::PanicHookInfo,)>>::call
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/alloc/src/boxed.rs:2289:9
  12:     0x7f68429759e8 - rustc_driver_impl[fdb3a64937eae85c]::install_ice_hook::{closure#1}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_driver_impl/src/lib.rs:1480:17
  13:     0x7f68429759e8 - <alloc[eced2805efacec9a]::boxed::Box<rustc_driver_impl[fdb3a64937eae85c]::install_ice_hook::{closure#1}> as core[d8f1845231c5d568]::ops::function::Fn<(&dyn for<'a, 'b> core[d8f1845231c5d568]::ops::function::Fn<(&'a std[170a660505e6ad19]::panic::PanicHookInfo<'b>,), Output = ()> + core[d8f1845231c5d568]::marker::Send + core[d8f1845231c5d568]::marker::Sync, &std[170a660505e6ad19]::panic::PanicHookInfo)>>::call
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/alloc/src/boxed.rs:2289:9
  14:     0x7f68429759e8 - std[170a660505e6ad19]::panicking::update_hook::<alloc[eced2805efacec9a]::boxed::Box<rustc_driver_impl[fdb3a64937eae85c]::install_ice_hook::{closure#1}>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:235:47
  15:     0x7f684626b41a - <alloc[eced2805efacec9a]::boxed::Box<dyn for<'a, 'b> core[d8f1845231c5d568]::ops::function::Fn<(&'a std[170a660505e6ad19]::panic::PanicHookInfo<'b>,), Output = ()> + core[d8f1845231c5d568]::marker::Send + core[d8f1845231c5d568]::marker::Sync> as core[d8f1845231c5d568]::ops::function::Fn<(&std[170a660505e6ad19]::panic::PanicHookInfo,)>>::call
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/alloc/src/boxed.rs:2289:9
  16:     0x7f684626b41a - std[170a660505e6ad19]::panicking::panic_with_hook
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:833:13
  17:     0x7f6846241002 - std[170a660505e6ad19]::panicking::panic_handler::{closure#0}
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:698:13
  18:     0x7f684622cae9 - std[170a660505e6ad19]::sys::backtrace::__rust_end_short_backtrace::<std[170a660505e6ad19]::panicking::panic_handler::{closure#0}, !>
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/sys/backtrace.rs:182:18
  19:     0x7f6846242f6d - __rustc[b1314c2b89442e1d]::rust_begin_unwind
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:689:5
  20:     0x7f68415209ec - core[d8f1845231c5d568]::panicking::panic_fmt
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/core/src/panicking.rs:80:14
  21:     0x7f684279b876 - rustc_const_eval[de7862fd2bcf1087]::const_eval::eval_queries::eval_to_allocation_raw_provider
  22:     0x7f68450f2432 - rustc_query_impl[96521f1206be213d]::query_impl::eval_to_allocation_raw::invoke_provider_fn::__rust_begin_short_backtrace
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/query_impl.rs:118:46
  23:     0x7f6844ea109e - rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:417:59
  24:     0x7f6844ea109e - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context::<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:56:9
  25:     0x7f6844ea109e - <std[170a660505e6ad19]::thread::local::LocalKey<core[d8f1845231c5d568]::cell::Cell<*const ()>>>::try_with::<rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/thread/local.rs:462:12
  26:     0x7f6844ea109e - <std[170a660505e6ad19]::thread::local::LocalKey<core[d8f1845231c5d568]::cell::Cell<*const ()>>>::with::<rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/thread/local.rs:426:20
  27:     0x7f6844ea109e - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context::<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:53:9
  28:     0x7f6844ea109e - rustc_query_impl[96521f1206be213d]::plumbing::start_query::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/plumbing.rs:74:9
  29:     0x7f6844ea109e - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context::<rustc_query_impl[96521f1206be213d]::plumbing::start_query<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:86:36
  30:     0x7f6844ea109e - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context_opt::<rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context<rustc_query_impl[96521f1206be213d]::plumbing::start_query<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:75:18
  31:     0x7f6844ea109e - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context::<rustc_query_impl[96521f1206be213d]::plumbing::start_query<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:86:5
  32:     0x7f6844ea109e - rustc_query_impl[96521f1206be213d]::plumbing::start_query::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/plumbing.rs:61:5
  33:     0x7f6844ea109e - rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:417:17
  34:     0x7f6844ea109e - rustc_query_impl[96521f1206be213d]::execution::try_execute_query::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>, false>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:325:17
  35:     0x7f68450f2804 - rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:623:32
  36:     0x7f68450f2804 - stacker[c93bf5ae587a613b]::maybe_grow::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  37:     0x7f68450f2804 - rustc_data_structures[b9d52ebac64c318a]::stack::ensure_sufficient_stack::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>::{closure#0}>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_data_structures/src/stack.rs:21:5
  38:     0x7f68450f2804 - rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:623:5
  39:     0x7f68450f2804 - rustc_query_impl[96521f1206be213d]::query_impl::eval_to_allocation_raw::execute_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/query_impl.rs:82:30
  40:     0x7f68427a2243 - rustc_middle[168fdf452f0a4ba3]::query::inner::query_get_at::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 24usize]>>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/query/inner.rs:45:17
  41:     0x7f68427a2243 - <rustc_middle[168fdf452f0a4ba3]::query::plumbing::TyCtxtAt>::eval_to_allocation_raw
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/query/plumbing.rs:574:46
  42:     0x7f68427a2243 - <rustc_middle[168fdf452f0a4ba3]::ty::context::TyCtxt>::eval_to_allocation_raw
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/query/plumbing.rs:562:39
  43:     0x7f68427a2243 - rustc_const_eval[de7862fd2bcf1087]::const_eval::valtrees::eval_to_valtree
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_const_eval/src/const_eval/valtrees.rs:252:27
  44:     0x7f6843a2bc9e - rustc_const_eval[de7862fd2bcf1087]::provide::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_const_eval/src/lib.rs:32:9
  45:     0x7f6843a2bc9e - <rustc_const_eval[de7862fd2bcf1087]::provide::{closure#0} as core[d8f1845231c5d568]::ops::function::FnOnce<(rustc_middle[168fdf452f0a4ba3]::ty::context::TyCtxt, rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>)>>::call_once
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/core/src/ops/function.rs:250:5
  46:     0x7f68450ac4f2 - rustc_query_impl[96521f1206be213d]::query_impl::eval_to_valtree::invoke_provider_fn::__rust_begin_short_backtrace
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/query_impl.rs:118:46
  47:     0x7f6844e9e514 - rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:417:59
  48:     0x7f6844e9e514 - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context::<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:56:9
  49:     0x7f6844e9e514 - <std[170a660505e6ad19]::thread::local::LocalKey<core[d8f1845231c5d568]::cell::Cell<*const ()>>>::try_with::<rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/thread/local.rs:462:12
  50:     0x7f6844e9e514 - <std[170a660505e6ad19]::thread::local::LocalKey<core[d8f1845231c5d568]::cell::Cell<*const ()>>>::with::<rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/thread/local.rs:426:20
  51:     0x7f6844e9e514 - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::enter_context::<rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:53:9
  52:     0x7f6844e9e514 - rustc_query_impl[96521f1206be213d]::plumbing::start_query::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/plumbing.rs:74:9
  53:     0x7f6844e9e514 - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context::<rustc_query_impl[96521f1206be213d]::plumbing::start_query<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:86:36
  54:     0x7f6844e9e514 - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context_opt::<rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context<rustc_query_impl[96521f1206be213d]::plumbing::start_query<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:75:18
  55:     0x7f6844e9e514 - rustc_middle[168fdf452f0a4ba3]::ty::context::tls::with_context::<rustc_query_impl[96521f1206be213d]::plumbing::start_query<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>::{closure#0}, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/ty/context/tls.rs:86:5
  56:     0x7f6844e9e514 - rustc_query_impl[96521f1206be213d]::plumbing::start_query::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/plumbing.rs:61:5
  57:     0x7f6844e9e514 - rustc_query_impl[96521f1206be213d]::execution::execute_job_non_incr::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:417:17
  58:     0x7f6844e9e514 - rustc_query_impl[96521f1206be213d]::execution::try_execute_query::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>, false>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:325:17
  59:     0x7f68450ac8b4 - rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:623:32
  60:     0x7f68450ac8b4 - stacker[c93bf5ae587a613b]::maybe_grow::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  61:     0x7f68450ac8b4 - rustc_data_structures[b9d52ebac64c318a]::stack::ensure_sufficient_stack::<rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>, rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>::{closure#0}>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_data_structures/src/stack.rs:21:5
  62:     0x7f68450ac8b4 - rustc_query_impl[96521f1206be213d]::execution::execute_query_non_incr_inner::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/execution.rs:623:5
  63:     0x7f68450ac8b4 - rustc_query_impl[96521f1206be213d]::query_impl::eval_to_valtree::execute_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_query_impl/src/query_impl.rs:82:30
  64:     0x7f6844006085 - rustc_middle[168fdf452f0a4ba3]::query::inner::query_get_at::<rustc_middle[168fdf452f0a4ba3]::query::caches::DefaultCache<rustc_middle[168fdf452f0a4ba3]::ty::PseudoCanonicalInput<rustc_middle[168fdf452f0a4ba3]::mir::interpret::GlobalId>, rustc_middle[168fdf452f0a4ba3]::query::erase::ErasedData<[u8; 16usize]>>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/query/inner.rs:45:17
  65:     0x7f6844006085 - <rustc_middle[168fdf452f0a4ba3]::query::plumbing::TyCtxtAt>::eval_to_valtree
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/query/plumbing.rs:574:46
  66:     0x7f6844006085 - <rustc_middle[168fdf452f0a4ba3]::ty::context::TyCtxt>::const_eval_global_id_for_typeck
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/mir/interpret/queries.rs:203:27
  67:     0x7f68440050cb - <rustc_middle[168fdf452f0a4ba3]::ty::context::TyCtxt>::const_eval_resolve_for_typeck
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_middle/src/mir/interpret/queries.rs:120:14
  68:     0x7f68442b743d - <rustc_mir_build[69f1187a66c64034]::thir::pattern::const_to_pat::ConstToPat>::unevaluated_to_pat
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs:111:38
  69:     0x7f6844259e3b - <rustc_mir_build[69f1187a66c64034]::thir::pattern::PatCtxt>::const_to_pat::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs:47:55
  70:     0x7f6844259e3b - <rustc_mir_build[69f1187a66c64034]::thir::pattern::PatCtxt>::const_to_pat
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs:36:5
  71:     0x7f68442469bf - <rustc_mir_build[69f1187a66c64034]::thir::pattern::PatCtxt>::lower_path
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/mod.rs:661:32
  72:     0x7f68442469bf - <rustc_mir_build[69f1187a66c64034]::thir::pattern::PatCtxt>::lower_pat_expr
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/mod.rs:693:51
  73:     0x7f68442440c1 - <rustc_mir_build[69f1187a66c64034]::thir::pattern::PatCtxt>::lower_pattern_unadjusted
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/mod.rs:332:54
  74:     0x7f68442440c1 - <rustc_mir_build[69f1187a66c64034]::thir::pattern::PatCtxt>::lower_pattern
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/mod.rs:134:23
  75:     0x7f68442d95c2 - rustc_mir_build[69f1187a66c64034]::thir::pattern::pat_from_hir::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/mod.rs:67:28
  76:     0x7f68442d95c2 - rustc_mir_build[69f1187a66c64034]::thir::pattern::pat_from_hir
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/pattern/mod.rs:46:1
  77:     0x7f684425559b - <rustc_mir_build[69f1187a66c64034]::thir::cx::ThirBuildCx>::pattern_from_hir_with_annotation
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/cx/mod.rs:120:9
  78:     0x7f684425559b - <rustc_mir_build[69f1187a66c64034]::thir::cx::ThirBuildCx>::pattern_from_hir
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/cx/mod.rs:112:14
  79:     0x7f684425559b - <rustc_mir_build[69f1187a66c64034]::thir::cx::ThirBuildCx>::convert_arm
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/cx/expr.rs:1171:27
  80:     0x7f684422d60b - <rustc_mir_build[69f1187a66c64034]::thir::cx::ThirBuildCx>::make_mirror_unadjusted::{closure#0}::{closure#15}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_mir_build/src/thir/cx/expr.rs:926:48
  
< snip >


 229:     0x7f68429677e5 - <scoped_tls[cf01fa775f28c1c3]::ScopedKey<rustc_span[417a2dbd7b74cf8a]::SessionGlobals>>::set::<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}, ()>
                               at /rust/deps/scoped-tls-1.0.1/src/lib.rs:137:9
 230:     0x7f68429677e5 - rustc_span[417a2dbd7b74cf8a]::create_session_globals_then::<(), rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_span/src/lib.rs:154:21
 231:     0x7f68429677e5 - rustc_interface[923575c5382274f]::util::run_in_thread_with_globals::<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/compiler/rustc_interface/src/util.rs:157:17
 232:     0x7f68429677e5 - std[170a660505e6ad19]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/sys/backtrace.rs:166:18
 233:     0x7f684297e827 - std[170a660505e6ad19]::thread::lifecycle::spawn_unchecked::<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/thread/lifecycle.rs:70:13
 234:     0x7f684297e827 - <core[d8f1845231c5d568]::panic::unwind_safe::AssertUnwindSafe<std[170a660505e6ad19]::thread::lifecycle::spawn_unchecked<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}> as core[d8f1845231c5d568]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/core/src/panic/unwind_safe.rs:275:9
 235:     0x7f684297e827 - std[170a660505e6ad19]::panicking::catch_unwind::do_call::<core[d8f1845231c5d568]::panic::unwind_safe::AssertUnwindSafe<std[170a660505e6ad19]::thread::lifecycle::spawn_unchecked<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:581:40
 236:     0x7f684297e827 - std[170a660505e6ad19]::panicking::catch_unwind::<(), core[d8f1845231c5d568]::panic::unwind_safe::AssertUnwindSafe<std[170a660505e6ad19]::thread::lifecycle::spawn_unchecked<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panicking.rs:544:19
 237:     0x7f684297e827 - std[170a660505e6ad19]::panic::catch_unwind::<core[d8f1845231c5d568]::panic::unwind_safe::AssertUnwindSafe<std[170a660505e6ad19]::thread::lifecycle::spawn_unchecked<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/panic.rs:359:14
 238:     0x7f684297e827 - std[170a660505e6ad19]::thread::lifecycle::spawn_unchecked::<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/thread/lifecycle.rs:68:26
 239:     0x7f684297e827 - <std[170a660505e6ad19]::thread::lifecycle::spawn_unchecked<rustc_interface[923575c5382274f]::util::run_in_thread_with_globals<rustc_interface[923575c5382274f]::util::run_in_thread_pool_with_globals<rustc_interface[923575c5382274f]::interface::run_compiler<(), rustc_driver_impl[fdb3a64937eae85c]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[d8f1845231c5d568]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /rustc-dev/99eed207b47aca1fec5c665531db8e948a92d0ca/library/core/src/ops/function.rs:250:5
 240:     0x7f6846278bcf - <alloc[eced2805efacec9a]::boxed::Box<dyn core[d8f1845231c5d568]::ops::function::FnOnce<(), Output = ()> + core[d8f1845231c5d568]::marker::Send> as core[d8f1845231c5d568]::ops::function::FnOnce<()>>::call_once
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/alloc/src/boxed.rs:2275:9
 241:     0x7f6846278bcf - <std[170a660505e6ad19]::sys::thread::unix::Thread>::new::thread_start
                               at /rustc/99eed207b47aca1fec5c665531db8e948a92d0ca/library/std/src/sys/thread/unix.rs:118:17
 242:     0x7f683fa9697a - <unknown>
 243:     0x7f683fb1a2bc - <unknown>
 244:                0x0 - <unknown>

error: the compiler unexpectedly panicked. This is a bug

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.97.0-nightly (99eed207b 2026-05-10) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [eval_to_allocation_raw] const-evaluating + checking `CONST`
#1 [eval_to_valtree] evaluating type-level constant
#2 [thir_body] building THIR for `main`
#3 [check_unsafety] unsafety-checking `main`
#4 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.

```

</p>
</details>

<!--
query stack:
#0 [eval_to_allocation_raw] const-evaluating + checking `CONST`
#1 [eval_to_valtree] evaluating type-level constant
#2 [thir_body] building THIR for `main`
#3 [check_unsafety] unsafety-checking `main`
#4 [analysis] running analysis passes on crate `mvce`

-->
@rustbot label +F-min_generic_const_args

--- a/compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs
+++ b/compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs
@@ -12,7 +12,9 @@ use rustc_infer::traits::Obligation;
 use rustc_middle::mir::interpret::ErrorHandled;
 use rustc_middle::span_bug;
 use rustc_middle::thir::{FieldPat, Pat, PatKind};
-use rustc_middle::ty::{self, Ty, TyCtxt, TypeSuperVisitable, TypeVisitableExt, TypeVisitor};
+use rustc_middle::ty::{
+    self, Ty, TyCtxt, TypeSuperVisitable, TypeVisitableExt, TypeVisitor, Unnormalized,
+};
 use rustc_span::def_id::DefId;
 use rustc_span::{DUMMY_SP, Span};
 use rustc_trait_selection::traits::ObligationCause;
@@ -106,72 +108,92 @@ impl<'tcx> ConstToPat<'tcx> {
             self.tcx.erase_and_anonymize_regions(self.typing_env).with_codegen_normalized(self.tcx);
         let uv = self.tcx.erase_and_anonymize_regions(uv);
 
-        // try to resolve e.g. associated constants to their definition on an impl, and then
-        // evaluate the const.
-        let valtree = match self.tcx.const_eval_resolve_for_typeck(typing_env, uv, self.span) {
-            Ok(Ok(c)) => c,
-            Err(ErrorHandled::Reported(_, _)) => {
-                // Let's tell the use where this failing const occurs.
-                let mut err =
-                    self.tcx.dcx().create_err(CouldNotEvalConstPattern { span: self.span });
-                // We've emitted an error on the original const, it would be redundant to complain
-                // on its use as well.
-                if let ty::ConstKind::Unevaluated(uv) = self.c.kind()
-                    && let ty::UnevaluatedConstKind::Projection { .. }
-                    | ty::UnevaluatedConstKind::Inherent { .. }
-                    | ty::UnevaluatedConstKind::Free { .. } = uv.kind
-                {
-                    err.downgrade_to_delayed_bug();
-                }
+        // FIXME(gca): This will become insufficient once associated constants can be
+        // implemented as `type` consts (project-const-generics#76). At that point it'll
+        // become necessary to just use type system normalization for all const patterns
+        // but that's not yet possible.
+        let mut thir_pat = if uv.kind.is_type_const(self.tcx) {
+            let Ok(normalize) = self
+                .tcx
+                .try_normalize_erasing_regions(self.typing_env, Unnormalized::new_wip(self.c))
+            else {
+                let err = self.tcx.dcx().create_err(CouldNotEvalConstPattern { span: self.span });
                 return self.mk_err(err, ty);
-            }
-            Err(ErrorHandled::TooGeneric(_)) => {
-                let mut e = self
-                    .tcx
-                    .dcx()
-                    .create_err(ConstPatternDependsOnGenericParameter { span: self.span });
-                for arg in uv.args {
-                    if let ty::GenericArgKind::Type(ty) = arg.kind()
-                        && let ty::Param(param_ty) = ty.kind()
+            };
+
+            let ty::ConstKind::Value(value) = normalize.kind() else {
+                let err = self.tcx.dcx().create_err(CouldNotEvalConstPattern { span: self.span });
+                return self.mk_err(err, ty);
+            };
+            self.valtree_to_pat(value)
+        } else {
+            // try to resolve e.g. associated constants to their definition on an impl, and then
+            // evaluate the const.
+            let valtree = match self.tcx.const_eval_resolve_for_typeck(typing_env, uv, self.span) {
+                Ok(Ok(c)) => c,
+                Err(ErrorHandled::Reported(_, _)) => {
+                    // Let's tell the use where this failing const occurs.
+                    let mut err =
+                        self.tcx.dcx().create_err(CouldNotEvalConstPattern { span: self.span });
+                    // We've emitted an error on the original const, it would be redundant to complain
+                    // on its use as well.
+                    if let ty::ConstKind::Unevaluated(uv) = self.c.kind()
+                        && let ty::UnevaluatedConstKind::Projection { .. }
+                        | ty::UnevaluatedConstKind::Inherent { .. }
+                        | ty::UnevaluatedConstKind::Free { .. } = uv.kind
                     {
-                        let def_id = self.tcx.hir_enclosing_body_owner(self.id);
-                        let generics = self.tcx.generics_of(def_id);
-                        let param = generics.type_param(*param_ty, self.tcx);
-                        let span = self.tcx.def_span(param.def_id);
-                        e.span_label(span, "constant depends on this generic parameter");
-                        if let Some(ident) = self.tcx.def_ident_span(def_id)
-                            && self.tcx.sess.source_map().is_multiline(ident.between(span))
+                        err.downgrade_to_delayed_bug();
+                    }
+                    return self.mk_err(err, ty);
+                }
+                Err(ErrorHandled::TooGeneric(_)) => {
+                    let mut e = self
+                        .tcx
+                        .dcx()
+                        .create_err(ConstPatternDependsOnGenericParameter { span: self.span });
+                    for arg in uv.args {
+                        if let ty::GenericArgKind::Type(ty) = arg.kind()
+                            && let ty::Param(param_ty) = ty.kind()
                         {
-                            // Display the `fn` name as well in the diagnostic, as the generic isn't
-                            // in the same line and it could be confusing otherwise.
-                            e.span_label(ident, "");
+                            let def_id = self.tcx.hir_enclosing_body_owner(self.id);
+                            let generics = self.tcx.generics_of(def_id);
+                            let param = generics.type_param(*param_ty, self.tcx);
+                            let span = self.tcx.def_span(param.def_id);
+                            e.span_label(span, "constant depends on this generic parameter");
+                            if let Some(ident) = self.tcx.def_ident_span(def_id)
+                                && self.tcx.sess.source_map().is_multiline(ident.between(span))
+                            {
+                                // Display the `fn` name as well in the diagnostic, as the generic isn't
+                                // in the same line and it could be confusing otherwise.
+                                e.span_label(ident, "");
+                            }
                         }
                     }
+                    return self.mk_err(e, ty);
                 }
-                return self.mk_err(e, ty);
-            }
-            Ok(Err(bad_ty)) => {
-                // The pattern cannot be turned into a valtree.
-                let e = match bad_ty.kind() {
-                    ty::Adt(def, ..) => {
-                        assert!(def.is_union());
-                        self.tcx.dcx().create_err(UnionPattern { span: self.span })
-                    }
-                    ty::FnPtr(..) | ty::RawPtr(..) => {
-                        self.tcx.dcx().create_err(PointerPattern { span: self.span })
-                    }
-                    _ => self.tcx.dcx().create_err(InvalidPattern {
-                        span: self.span,
-                        non_sm_ty: bad_ty,
-                        prefix: bad_ty.prefix_string(self.tcx).to_string(),
-                    }),
-                };
-                return self.mk_err(e, ty);
-            }
-        };
+                Ok(Err(bad_ty)) => {
+                    // The pattern cannot be turned into a valtree.
+                    let e = match bad_ty.kind() {
+                        ty::Adt(def, ..) => {
+                            assert!(def.is_union());
+                            self.tcx.dcx().create_err(UnionPattern { span: self.span })
+                        }
+                        ty::FnPtr(..) | ty::RawPtr(..) => {
+                            self.tcx.dcx().create_err(PointerPattern { span: self.span })
+                        }
+                        _ => self.tcx.dcx().create_err(InvalidPattern {
+                            span: self.span,
+                            non_sm_ty: bad_ty,
+                            prefix: bad_ty.prefix_string(self.tcx).to_string(),
+                        }),
+                    };
+                    return self.mk_err(e, ty);
+                }
+            };
 
-        // Lower the valtree to a THIR pattern.
-        let mut thir_pat = self.valtree_to_pat(ty::Value { ty, valtree });
+            // Lower the valtree to a THIR pattern.
+            self.valtree_to_pat(ty::Value { ty, valtree })
+        };
 
         if !thir_pat.references_error() {
             // Always check for `PartialEq` if we had no other errors yet.

--- a/tests/ui/const-generics/mgca/type_const_in_pattern.rs
+++ b/tests/ui/const-generics/mgca/type_const_in_pattern.rs
@@ -1,0 +1,28 @@
+//@ check-pass
+#![feature(min_generic_const_args)]
+#![expect(incomplete_features)]
+#![allow(irrefutable_let_patterns)]
+
+type const FOO: usize = 1_usize;
+
+struct Inherent;
+
+impl Inherent {
+    type const BAR: usize = 1_usize;
+}
+
+trait Trait {
+    type const BAZ: usize;
+}
+
+struct Assoc;
+
+impl Trait for Assoc {
+    type const BAZ: usize = 1_usize;
+}
+
+fn main() {
+    if let FOO = 1 {}
+    if let Inherent::BAR = 1 {}
+    if let <Assoc as Trait>::BAZ = 1 {}
+}

--- a/tests/ui/const-generics/mgca/type_const_in_pattern.rs
+++ b/tests/ui/const-generics/mgca/type_const_in_pattern.rs
@@ -3,7 +3,7 @@
 #![expect(incomplete_features)]
 #![allow(irrefutable_let_patterns)]
 
-type const FOO: usize = 1_usize;
+type const CONST: usize = 1_usize;
 
 struct Inherent;
 
@@ -22,7 +22,12 @@ impl Trait for Assoc {
 }
 
 fn main() {
-    if let FOO = 1 {}
+    if let CONST = 1 {}
     if let Inherent::BAR = 1 {}
     if let <Assoc as Trait>::BAZ = 1 {}
+
+    match CONST {
+        CONST => 0,
+        _ => 1,
+    };
 }

--- a/tests/ui/const-generics/mgca/type_const_in_pattern_too_generic.rs
+++ b/tests/ui/const-generics/mgca/type_const_in_pattern_too_generic.rs
@@ -1,0 +1,13 @@
+#![feature(min_generic_const_args)]
+#![expect(incomplete_features)]
+
+trait Trait {
+    type const ASSOC: usize;
+}
+
+fn test<T: Trait>() {
+    if let <T as Trait>::ASSOC = 1 {}
+    //~^ ERROR could not evaluate constant pattern
+}
+
+fn main() {}

--- a/tests/ui/const-generics/mgca/type_const_in_pattern_too_generic.stderr
+++ b/tests/ui/const-generics/mgca/type_const_in_pattern_too_generic.stderr
@@ -1,0 +1,13 @@
+error: could not evaluate constant pattern
+  --> $DIR/type_const_in_pattern_too_generic.rs:9:12
+   |
+LL | trait Trait {
+   | -----------
+LL |     type const ASSOC: usize;
+   |     ----------------------- constant defined here
+...
+LL |     if let <T as Trait>::ASSOC = 1 {}
+   |            ^^^^^^^^^^^^^^^^^^^ could not evaluate constant
+
+error: aborting due to 1 previous error
+

